### ISSUE TITLE
1. "GPU Conversion" setting not selected in UVR 5.50! 2. "Ensemble Mode" settings are available in UVR 5.40, but not in UVR 5.50.

### ISSUE THREAD

**Issue by Cuarizma2020** [Anjok07/ultimatevocalremovergui#330]
1. "GPU Conversion" setting not selected in UVR 5.50!
2. "Ensemble Mode" settings are available in UVR 5.40, but not in UVR 5.50.

**Comment by Anjok07** [comment]
1. If the "GPU Conversion" check button is not selectable, it means the application does not detect CUDA on your system. Previous versions of UVR allowed you to check the button, even if CUDA wasn't available, but would still default to the CPU if it wasn't available. The reason it disables 
2. "Ensemble Mode" settings have actually been *expanded* in UVR 5.5.0. Please see the change log for ensemble mode [here](https://github.com/Anjok07/ultimatevocalremovergui#ensemble-mode)

**Comment by 8bit-coder** [comment]
Well, actually I've had the same issue: GPU conversion was unelectable even though I had an Nvidia GTX 550 Ti installed in my system. The graphics card clearly supports CUDA, yet UVR5 didn't let me select the GPU Conversion option. Is there any way around this?

**Comment by Anjok07** [comment]
That GPU is not supported by Pytorch

**Comment by Anjok07** [comment]
> Well, actually I've had the same issue: GPU conversion was unelectable even though I had an Nvidia GTX 550 Ti installed in my system. The graphics card clearly supports CUDA, yet UVR5 didn't let me select the GPU Conversion option. Is there any way around this? 

That GPU is not supported by the version of Pytorch UVR requires to run most of these models. I can also confirm that GPU does not have enough V-RAM to run a vast majority of UVR's models.

**Comment by 8bit-coder** [comment]
> That GPU is not supported by the version of Pytorch UVR requires to run most of these models. I can also confirm that GPU does not have enough V-RAM to run a vast majority of UVRs models.

What about a GTX 750 Ti? Would that one be supported?

**Comment by Anjok07** [comment]
That GPU is not supported.

However, the GTX 1060 6GB is the minimum requirement to run the version of PyTorch that comes with UVR.

The GTX 1050ti has been reported to work, but the GTX 1060 6GB is the safest bet in terms of minimum requirements.

**Comment by 8bit-coder** [comment]
Dang, that sucks. Is it possible to replace the version of PyTorch the program comes bundled with an older version that DOES support CUDA 5.0?

**Comment by bootluca** [comment]
> 1. If the "GPU Conversion" check button is not selectable, it means the application does not detect CUDA on your system. Previous versions of UVR allowed you to check the button, even if CUDA wasn't available, but would still default to the CPU if it wasn't available. It disables when Cuda isn't available to make it more apparent to the user.
> 2. "Ensemble Mode" settings have actually been _expanded_ in UVR 5.5.0. Please see the change log for ensemble mode [here](https://github.com/Anjok07/ultimatevocalremovergui#ensemble-mode)

The problem is that without the "GPU conversion" selected, the model works worse. At this point I uninstall the upgrade and I reinstall the last version that allowed me to select that option which worked better

**Comment by Anjok07** [comment]
> > 1. If the "GPU Conversion" check button is not selectable, it means the application does not detect CUDA on your system. Previous versions of UVR allowed you to check the button, even if CUDA wasn't available, but would still default to the CPU if it wasn't available. It disables when Cuda isn't available to make it more apparent to the user.
> > 2. "Ensemble Mode" settings have actually been _expanded_ in UVR 5.5.0. Please see the change log for ensemble mode [here](https://github.com/Anjok07/ultimatevocalremovergui#ensemble-mode)
> 
> The problem is that without the "GPU conversion" selected, the model works worse. At this point I uninstall the upgrade and I reinstall the last version that allowed me to select that option which worked better

If you don't have a Nvidia GPU that's a 1060 or above, the "GPU Conversion" checkbox does **absolutely nothing**. Which GPU are you using?

Trust me, I wrote the code for v5.4 and v5.5, so I know. Which model are you using that works worse? I highly recommend you stay on v5.5 since the older versions have bugs and many of the newer models will not work correctly in v5.4 and below.

**Comment by bootluca** [comment]
> > > 1. If the "GPU Conversion" check button is not selectable, it means the application does not detect CUDA on your system. Previous versions of UVR allowed you to check the button, even if CUDA wasn't available, but would still default to the CPU if it wasn't available. It disables when Cuda isn't available to make it more apparent to the user.
> > > 2. "Ensemble Mode" settings have actually been _expanded_ in UVR 5.5.0. Please see the change log for ensemble mode [here](https://github.com/Anjok07/ultimatevocalremovergui#ensemble-mode)
> > 
> > 
> > The problem is that without the "GPU conversion" selected, the model works worse. At this point I uninstall the upgrade and I reinstall the last version that allowed me to select that option which worked better
> 
> If you don't have a Nvidia GPU that's a 1060 or above, the "GPU Conversion" checkbox does **absolutely nothing**. Which GPU are you using?
> 
> Trust me, I wrote the code for v5.4 and v5.5, so I know. Which model are you using that works worse? I highly recommend you stay on v5.5 since the older versions have bugs and many of the newer models will not work correctly in v5.4 and below.

Thanks for answering
Radeon RX 560
Method VR Architecture model 5_HP-karaoke-UVR

**Comment by Anjok07** [comment]
> > > > 1. If the "GPU Conversion" check button is not selectable, it means the application does not detect CUDA on your system. Previous versions of UVR allowed you to check the button, even if CUDA wasn't available, but would still default to the CPU if it wasn't available. It disables when Cuda isn't available to make it more apparent to the user.
> > > > 2. "Ensemble Mode" settings have actually been _expanded_ in UVR 5.5.0. Please see the change log for ensemble mode [here](https://github.com/Anjok07/ultimatevocalremovergui#ensemble-mode)
> > > 
> > > 
> > > The problem is that without the "GPU conversion" selected, the model works worse. At this point I uninstall the upgrade and I reinstall the last version that allowed me to select that option which worked better
> > 
> > If you don't have a Nvidia GPU that's a 1060 or above, the "GPU Conversion" checkbox does **absolutely nothing**. Which GPU are you using?
> > 
> > Trust me, I wrote the code for v5.4 and v5.5, so I know. Which model are you using that works worse? I highly recommend you stay on v5.5 since the older versions have bugs and many of the newer models will not work correctly in v5.4 and below.
> 
> Thanks for answering
> Radeon RX 560
> Method VR Architecture model 5_HP-karaoke-UVR
> 
> 

AMD GPU's are not supported with PyTorch in Windows. I can confirm 100% that the "GPU Conversion" button is doing absolutely nothing for you in v5.4, and the application is still defaulting to the CPU because it does not detect a compatible GPU. I only made it to disable the button in v5.5 to make the GPU in compatibility more apparent to the user (this was solely a cosmetic change).

Are your results coming out poorer in v5.5 with this model? If so, I can double-check to make sure the correct model parameters are set.

**Comment by bootluca** [comment]
> > > > > 1. If the "GPU Conversion" check button is not selectable, it means the application does not detect CUDA on your system. Previous versions of UVR allowed you to check the button, even if CUDA wasn't available, but would still default to the CPU if it wasn't available. It disables when Cuda isn't available to make it more apparent to the user.
> > > > > 2. "Ensemble Mode" settings have actually been _expanded_ in UVR 5.5.0. Please see the change log for ensemble mode [here](https://github.com/Anjok07/ultimatevocalremovergui#ensemble-mode)
> > > > 
> > > > 
> > > > The problem is that without the "GPU conversion" selected, the model works worse. At this point I uninstall the upgrade and I reinstall the last version that allowed me to select that option which worked better
> > > 
> > > 
> > > If you don't have a Nvidia GPU that's a 1060 or above, the "GPU Conversion" checkbox does **absolutely nothing**. Which GPU are you using?
> > > Trust me, I wrote the code for v5.4 and v5.5, so I know. Which model are you using that works worse? I highly recommend you stay on v5.5 since the older versions have bugs and many of the newer models will not work correctly in v5.4 and below.
> > 
> > 
> > Thanks for answering
> > Radeon RX 560
> > Method VR Architecture model 5_HP-karaoke-UVR
> 
> AMD GPU's are not supported with PyTorch in Windows. I can confirm 100% that the "GPU Conversion" button is doing absolutely nothing for you in v5.4, and the application is still defaulting to the CPU because it does not detect a compatible GPU. I only made it to disable the button in v5.5 to make the GPU incompatibility more apparent to the user (this was solely a cosmetic change).
> 
> Are your results coming out poorer in v5.5 with this model? I double-checked the parameters are set exactly the same in v5.4 and v5.5 for that model.

I no longer have the old version installed I can't check, but it seems to me that with the old version there were also other selectable options, besides GPU conversion, that improved the result, options that are no longer available on this version for that model, options that could be selected and that gave excellent results... perhaps something like TTA, I'm not sure
Alternatively could you tell me what is the best way to get a karaoke track and a main vocal track?

**Comment by Anjok07** [comment]
TTA is still an option. The VR Arch settings have been expanded. To get those options, go to the "Advanced VR Settings" menu via the Settings menu or by Right-clicking within the application and selecting "Advanced VR Settings". Regarding the best settings, you'll need to test out different settings as each song processes differently.

**Comment by Anjok07** [comment]
> Dang, that sucks. Is it possible to replace the version of PyTorch the program comes bundled with an older version that DOES support CUDA 5.0?

No, there is not. If there was, more than half the models would not work.

**Comment by bootluca** [comment]
> TTA is still an option. The VR Arch settings have been expanded. To get those options, go to the "Advanced VR Settings" menu via the Settings menu or by Right-clicking within the application and selecting "Advanced VR Settings". 

OK thanks a lot, this way I like the program again

**Comment by bootluca** [comment]
in the old version there was an option that allowed the software to make 2 passes before saving the instrumentals and vocals tracks, what was it?

**Comment by Anjok07** [comment]
> in the old version there was an option that allowed the software to make 2 passes before saving the instrumentals and vocals tracks, what was it?

Ensemble Mode

**Comment by Anjok07** [comment]
> > > in the old version there was an option that allowed the software to make 2 passes before saving the instrumentals and vocals tracks, what was it?
> > 
> > Ensemble Mode
> 
> no, never used ensemble mode, I meant in the VR 5_HP-karaoke model
> I think I understand, in version 5.4 there were 2 processes called "processing XX slices" and "processing XX+1 slices" which in the updated version I think have been merged into one big process called 
> 

All of that still happens on the backend. The console text in v5.5 isn't as verbose as v5.4 (for good reason).

**Comment by bootluca** [comment]
yes yes, I understand, I was getting worse results because I was not activating the options that compared to the old version are no longer on the main page but are more hidden. Now that I have activated them the results are the same as before. Also, I am curious to try the "high end process" option which was not present in the old version
Thanks
Luca

**Comment by qnz001** [comment]
Hi all,
I'm very new to this community and software world but learning from smart guys like you. From this discussion, it seems like my desktop with old Quadro 4000 (2GB RAM) will not work with UVR5.5. I'm thinking of installing the Nvidia Tesla P40 ( 24GB vRAM - after some tweaking with the cooling system etc). Will it work in UVR? Please forgive me if I asked a dumb question. Much obliged in advance.
